src/Features/Lsif/Generator/Graph/Capabilities.cs

```diff
@@ -35,6 +37,9 @@ internal sealed class Capabilities : Vertex
         [JsonProperty("diagnosticProvider")]
         public bool DiagnosticProvider { get; }
 
+        [JsonProperty("semanticTokensProvider")]
+        public string[]? SemanticTokensProvider { get; }
```

Comment:
Instead of one big array, should we make it an object with the list of token types as a property? Intention would be to enable us to better expand the capabilities in the future if we felt it warranted

ALso, why not `IReadOnlyList<string>`?

Reply:
Ah, good idea! I also realized that we weren't including tokenModifiers. Roslyn de facto doesn't support this but it's part of the LSP spec and should probably be included in LSIF.